File: server/app/schemas/notification.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict

from app.models.notification import NotificationType


class NotificationResponse(BaseModel):
    id: int
    type: NotificationType
    title: str
    message: str
    is_read: bool
    created_at: datetime
    read_at: datetime | None

    model_config = ConfigDict(from_attributes=True)


class NotificationPage(BaseModel):
    items: list[NotificationResponse]
    page: int
    page_size: int
    total: int
    pages: int
    unread_count: int

File: client/src/types/notification.ts
export type NotificationType = 'profile_submitted' | 'profile_verified' | 'profile_rejected' | 'account_suspended' | 'account_reactivated' | 'recommendation_ready'

export interface Notification {
  id: number
  type: NotificationType
  title: string
  message: string
  is_read: boolean
  created_at: string
  read_at: string | null
}

export interface NotificationPage {
  items: Notification[]
  page: number
  page_size: number
  total: number
  pages: number
  unread_count: number
}
